src/osd/PG.cc

```diff
@@ -3428,24 +2351,9 @@ bool PG::try_flush_or_schedule_async()
 ostream& operator<<(ostream& out, const PG& pg)
 {
   out << pg.recovery_state;
-  if (pg.scrubber.must_repair)
-    out << " MUST_REPAIR";
-  if (pg.scrubber.auto_repair)
-    out << " AUTO_REPAIR";
-  if (pg.scrubber.check_repair)
-    out << " CHECK_REPAIR";
-  if (pg.scrubber.deep_scrub_on_error)
-    out << " DEEP_SCRUB_ON_ERROR";
-  if (pg.scrubber.must_deep_scrub)
-    out << " MUST_DEEP_SCRUB";
-  if (pg.scrubber.must_scrub)
-    out << " MUST_SCRUB";
-  if (pg.scrubber.time_for_deep)
-    out << " TIME_FOR_DEEP";
-  if (pg.scrubber.need_auto)
-    out << " NEED_AUTO";
-  if (pg.scrubber.req_scrub)
-    out << " REQ_SCRUB";
+
+  // RRR listing flags_. Should we list 'planned_scrub' flags instead?
+  out << *pg.m_scrubber;
```

Comment:
@athanatos (@dzafman) to clarify:
in the suggested code we have two sets of flags (see discussion above). One set is the 'active' one: relevant only when we are scrubbing, and affects the running scrub. That set is initialized from the _other_ set - the 'planned next scrub' flag set.
m_planned_scrub is set by various conditions, and is cleared once used to initialize the 'active' flags.

My suggestion: output the active set when scrub is active; log the 'planned' set otherwise; output both for special occasions.
A minor issue: can I modify the flags text somewhat to clarify what set of flags is logged? otherwise, there's a problem with a couple of flags that appear in both sets.